internal/datacoord/compaction_trigger.go

```diff
@@ -693,6 +693,39 @@ func (t *compactionTrigger) ShouldCompactExpiry(fromTs uint64, compactTime *comp
 	return false
 }
 
+func getExpirQuantilesIndexByRatio(ratio float64, percentilesLen int) int {
+	// expirQuantiles is [20%, 40%, 60%, 80%, 100%] (len = 5).
+	// We map ratio to the nearest lower 20% bucket:
+	// 0~0.39 -> 20%, 0.4~0.59 -> 40%, 0.6~0.79 -> 60%, 0.8~0.99 -> 80%, >=1.0 -> 100%
+	if percentilesLen <= 0 {
+		return 0
+	}
+	step := 0.2
+	idx := int(ratio/step) - 1
```

Comment:
actually
0.6 = 0.5999999
0.6/0.2 = 2.9999999999999996
idx = int(2.99999) -1 = 1   instead of 2